polkadot/node/service/src/overseer.rs

```diff
@@ -145,39 +119,67 @@ where
 	pub notification_services: HashMap<PeerSet, Box<dyn NotificationService>>,
 }
 
-/// Obtain a prepared `OverseerBuilder`, that is initialized
-/// with all default values.
-pub fn prepared_overseer_builder<Spawner, RuntimeClient>(
+pub struct ExtendedOverseerGenArgs {
+	/// The underlying key value store for the parachains.
+	pub parachains_db: Arc<dyn polkadot_node_subsystem_util::database::Database>,
+	/// Configuration for the candidate validation subsystem.
+	pub candidate_validation_config: Option<CandidateValidationConfig>,
+	/// Configuration for the availability store subsystem.
+	pub availability_config: AvailabilityConfig,
+	/// POV request receiver.
+	pub pov_req_receiver: IncomingRequestReceiver<request_v1::PoVFetchingRequest>,
+	/// Erasure chunks request receiver.
+	pub chunk_req_receiver: IncomingRequestReceiver<request_v1::ChunkFetchingRequest>,
+	/// Enable PVF pre-checking
+	pub pvf_checker_enabled: bool,
```

Comment:
I think this flag is used to distinguish validators from collators, but we have `is_parachain_node` as well, which is even more descriptive. So maybe remove it altogether, unless I'm missing something?

Reply:
Hmmm, you're right, that flag probably made more sense before paritytech/polkadot#7396 somehow, but now it looks completely redundant. Will remove it.